tensorflow/compiler/jit/xla_gpu_device.cc

```diff
@@ -83,14 +83,16 @@ Status XlaGpuDeviceFactory::CreateDevices(
   }
   string allowed_gpus =
       session_options.config.gpu_options().visible_device_list();
+  auto parsed_gpus=ParseVisibleDeviceList(allowed_gpus).ValueOrDie();
+  // We want to fill the gpu_ids set with all devices if config string is empty.
   std::set<int> gpu_ids;
```

Comment:
If we're going to have ParseVisibleDeviceList return an optional, could we have just one variable?

```
absl::optional<std::set<int>> gpu_ids = ParseVisibleDeviceList(allowed_gpus);
if (!gpu_ids) {
  gpu_ids.emplace();
  for ... { gpu_ids->insert(i); }
}
for (int i : *gpu_ids) { ... }
```

Reply:
I thought of this as well. But it is not much different that current implementation since if the string is empty, function will return a nullopt so we have to instantiate new set, put it in optional and fill it with all devices and then iterate over it. I can also do that but this way seemed cleaner.

Reply:
> so we have to instantiate new set, put it in optional and fill it with all devices and then iterate over it. I can also do that but this way seemed cleaner.

That's what the code I suggested above does, right?

Reply:
Done.